src/js/libcs/OpImageDrawing.js

```diff
@@ -388,3 +388,167 @@ evaluator.cameravideo$0 = function() {
+            if (!helpercanvas) {
+                //creating helpercanvas only once increases the running time
+                helpercanvas = /** @type {HTMLCanvasElement} */ (document.createElement("canvas"));
+            }
+            helpercanvas.width = width;
+            helpercanvas.height = height;
+
+            ctx = helpercanvas.getContext('2d');
+            ctx.drawImage(img.img, x, y, width, height, 0, 0, width, height);
+            data = ctx.getImageData(0, 0, width, height).data;
+        }
+        for (var i in data) res.push(data[i] / 255);
+    }
+    return res;
+}
+
+/**
+ * imagergba(‹image›,x,y) implements imagergb(‹imagename›,x,y) from Cinderella, i.e.
+ * returns a 4 component vector ranging from (0-255, 0-255, 0-255, 0-1)
+ */
+evaluator.imagergba$3 = function(args, modifs) {
+    var img = imageFromValue(evaluateAndVal(args[0]));
+    var x = evaluateAndVal(args[1]);
+    var y = evaluateAndVal(args[2]);
+
+    if (!img || x.ctype !== 'number' || y.ctype !== 'number') return nada;
+
+    x = Math.round(x.value.real);
+    y = Math.round(y.value.real);
+    if (!Number.isFinite(x) || !Number.isFinite(y)) return nada;
+
+    var rgba = readPixelsIndirection(img, x, y, 1, 1);
+    return List.realVector([rgba[0] * 255, rgba[1] * 255, rgba[2] * 255, rgba[3]]);
+};
+
+evaluator.imagergb$3 = evaluator.imagergba$3; //According to reference
+
+/**
+ * imagergba(<point1>, <point2>, ‹image›, <point3>) returns the color at the coordinate
+ * <point3> assuming that the left/right lower corner is <point1>/<point2> resp.
+ */
+evaluator.imagergba$4 = function(args, modifs) {
+    var img = imageFromValue(evaluateAndVal(args[2]));
+
+    var interpolate = true; //default values
+    var repeat = false;
+
+    function handleModifs() {
+        var erg;
+        if (modifs.interpolate !== undefined) {
+            erg = evaluate(modifs.interpolate);
+            if (erg.ctype === 'boolean') {
+                interpolate = (erg.value);
+            }
+        }
+
+        if (modifs.repeat !== undefined) {
+            erg = evaluate(modifs.repeat);
+            if (erg.ctype === 'boolean') {
+                repeat = (erg.value);
+            }
+        }
+    }
+    handleModifs();
+
+    if (!img) return nada;
+
+    var w = img.width;
+    var h = img.height;
+
+    var w0 = evaluateAndHomog(args[0]);
+    var w1 = evaluateAndHomog(args[1]);
+    var v0 = evaluateAndHomog(List.realVector([0, h, 1]));
+    var v1 = evaluateAndHomog(List.realVector([w, h, 1]));
+
+    if (w0 === nada || w1 === nada || p === nada) return nada;
+
+    //create an orientation-reversing similarity transformation that maps w0->v0, w1->v1
+    var ii = List.ii;
+    var jj = List.jj;
+
+    var m1 = eval_helper.basismap(v0, v1, ii, jj); //interchange I and J,
+    var m2 = eval_helper.basismap(w0, w1, jj, ii); //see Thm. 18.4 of Perspectives on Projective Geometry
+    var p = evaluateAndHomog(args[3]);
```

Comment:
I wonder whether it would be a good idea to split the function at this point. Up till here you essentially computed the pixel position. Now you retrieve the actual data. If we at some point decide to support interpolation for the `imagergb$3` call, or to decide `imagergb$5` and so on for more complicated image mapping operations, then having specific position combinations and then sharing the interpolation would be very useful. Modifier handling would go into the second part. What do you think? We can keep this for a later change if you prefer.